src/ws/cockpitauth.c

```diff
@@ -633,8 +633,12 @@ reply_authorize_challenge (CockpitSession *session)
 
   if (response && !cookie)
     {
-      cockpit_memory_clear (session->authorization, -1);
-      g_free (session->authorization);
+      if (session->authorization)
+        {
```

Comment:
Hmm, why didn't you just keep the function as being NULL safe?

Reply:
The idea of a function that takes NULL/-1 is pretty unprecedented.  All of the call sites that took advantage of this functionality had their readability improved by open-coding the check explicitly.

I'm actually reconsidering keeping this function at all and rather replacing it with:

`explicit_bzero (x, strlen (x))`

at each `-1` use (and just plain `explicit_bzero` in the others).

Reply:
> The idea of a function that takes NULL/-1 is pretty unprecedented.

`free()`, `g_clear_error()` etc. all do that, for IMHO good reason. But anyway, while I like defensive functions, this isn't a veto. Dropping the (now rather shallow) `cockpit_memory_clear()` function wrapper entirely sounds good to me.